app/views/users/index.html.erb

```diff
@@ -1 +1,12 @@
-<h1><%= @users %></h1>
+<h1 class="users font-bold text-3xl">A list of all users </h1>
+<br>
+<ul>
+    <% @users.each do |user| %>
+        <li class="border mb-4 hover:bg-slate-200">
+           <%= link_to user_path(user) do %>
+                <%= render "shared/user", user: user %>
+           <% end %>
+
+        </li>
+    <% end %>
```

Comment:
- [ ] [OPTIONAL] You can create `app/views/users/_user.html.erb` (use partial) and divide your code for best practices 
e.g:
```ruby
    # app/views/users/_user.html.erb

     #   HTML Code Here displaying
     #   user.photo, user.name and user.posts_counter

    # app/views/users/index.html.erb

    <%= render @users %>
```